src/utils/hooks/use-animation.ts

```diff
@@ -0,0 +1,53 @@
+import { useRef, useEffect } from 'react'
+import { Animated } from 'react-native'
+
+export const useAnimatedValue = (initialValue: number): Animated.Value => {
+  const ref = useRef(new Animated.Value(initialValue))
+  return ref.current
+}
+
+interface BaseAnimationConfig {
+  initialValue?: number
+}
+
+export type TimingAnimationConfig = BaseAnimationConfig &
+  ({ type: 'timing' } & Animated.TimingAnimationConfig)
+
+export type SpringAnimationConfig = BaseAnimationConfig &
+  ({ type: 'spring' } & Animated.SpringAnimationConfig)
+
+export type UseAnimationConfig = TimingAnimationConfig | SpringAnimationConfig
+
+const getInitialValue = (config: UseAnimationConfig): number => {
+  if (typeof config.initialValue !== 'undefined') {
+    return config.initialValue
+  } else {
+    return config.toValue as number // TODO deal with other types possibilities here
```

Comment:
What are the other possible types?

Reply:
`toValue` type here comes from React Native `Animated` API, which can receive other `Animated.Value`, `Animated.Interpolation`,  `Animated.ValueXY` objects in `toValue`.

However, I updated the code so `toValue` forces `number` type and removed the cast. Passing `Animated` values is not 
supported in this hook and should be considered a most advanced Animation usage.